balancer/catalyst/catalyst_balancer_test.go

```diff
@@ -0,0 +1,346 @@
+package catalyst
+
+import (
+	"context"
+	"github.com/livepeer/catalyst-api/cluster"
+	"golang.org/x/sync/errgroup"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var CPUOverloadedNode = ScoredNode{
+	Node: Node{Name: "cpu_overload"},
+	NodeMetrics: NodeMetrics{
+		CPUUsagePercentage:       100,
+		RAMUsagePercentage:       0,
+		BandwidthUsagePercentage: 0,
+	},
+	Streams: map[string]Stream{},
+}
+
+var RAMOverloadedNode = ScoredNode{
+	Node: Node{Name: "mem_overload"},
+	NodeMetrics: NodeMetrics{
+		CPUUsagePercentage:       0,
+		RAMUsagePercentage:       100,
+		BandwidthUsagePercentage: 0,
+	},
+	Streams: map[string]Stream{},
+}
+
+var BandwidthOverloadedNode = ScoredNode{
+	Node: Node{Name: "bw_overload"},
+	NodeMetrics: NodeMetrics{
+		CPUUsagePercentage:       0,
+		RAMUsagePercentage:       0,
+		BandwidthUsagePercentage: 100,
+	},
+	Streams: map[string]Stream{},
+}
+
+func TestItReturnsItselfWhenNoOtherNodesPresent(t *testing.T) {
+	// Make the node handling the request unfavourable in terms of stats, to make sure
+	// it'll still pick itself if it's the only option
+	_, err := SelectNode([]ScoredNode{}, "some-stream-id", 0, 0)
+	require.EqualError(t, err, "no nodes to select from")
```

Comment:
This seems to contradict the test title - do we want to return ourself or that it errors out?

Reply:
Ah whoops yeah looks like we meant to come back to this. Returning ourself seems like a better option if that's possible, the balancer could have an emergency `SelfNodeName` to return if nothing else is found?